peepingtom/visualisation/depictors/particles.py

```diff
@@ -1,34 +1,35 @@
 import numpy as np
 from napari.layers import Points, Vectors
 
-from ..base import Depictor
+from .base import Depictor
 
 
 class ParticleDepictor(Depictor):
-    def make_layers(self, point_kwargs={}, vector_kwargs={}):
+    def init_layers(self, point_kwargs={}, vector_kwargs={}):
         pkwargs = {'size': 3}
         vkwargs = {'length': 10}
 
         pkwargs.update(point_kwargs)
         vkwargs.update(vector_kwargs)
 
-        p_layer = Points(self.datablock.positions.zyx,
-                         name=f'{self.name} - particle positions',
-                         properties=self.datablock.properties.data,
-                         **pkwargs)
+        p_layer = self.make_points_layer(self.datablock.positions.as_zyx(),
+                                         name=f'{self.name} - particle positions',
+                                         properties=self.datablock.properties.data,
+                                         **pkwargs)
         self.layers.append(p_layer)
 
         # get positions and 'projection' vectors
-        positions = self.datablock.positions.zyx
+        positions = self.datablock.positions.as_zyx()
         unit_z_rotated_order_xyz = self.datablock.orientations.oriented_vectors('z').reshape((-1, 3))
         unit_z_rotated_order_zyx = unit_z_rotated_order_xyz[:, ::-1]
-
-        napari_vectors = np.stack([positions,
-                                   unit_z_rotated_order_zyx],
-                                  axis=1)
-        v_layer = Vectors(napari_vectors,
-                          name=f'{self.name} - particle orientations',
-                          **vkwargs)
+        # attach appropriate higher dimensions indeces to vectors
+        if self.datablock.positions.ndim > 3:
+            unit_z_rotated_order_zyx = np.concatenate([positions[:, :-3], unit_z_rotated_order_zyx], axis=1)
```

Comment:
why do you reverse the ordering of positions here? this is already done in the .as_zyx method, no?

Reply:
hmm, I seem to have misread this... not sure why it's broken